utils/cp_utils.py

```diff
@@ -37,34 +46,85 @@ def run_cellprofiler(
         sqlite_name (str, optional): string with name for SQLite file for an analysis pipeline (default is None)
         analysis_run (bool, optional): will use functions to complete an analysis pipeline (default is False)
     """
-    # run CellProfiler on a plate that has not been analyzed yet
-    command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} --data-file {path_to_loaddata}"
-    os.system(command)
+    # check to make sure the paths to files are correct and they exists before running CellProfiler
+    if not os.path.exists(path_to_pipeline):
+        raise FileNotFoundError(f"Directory '{path_to_pipeline}' does not exist")
+    if not os.path.exists(path_to_loaddata):
+        raise FileNotFoundError(f"Directory '{path_to_loaddata}' does not exist")
+
+    if not analysis_run:
+        # run CellProfiler on a plate that has not been analyzed yet
+        print(f"Starting CellProfiler run on {pathlib.Path(path_to_loaddata).name}")
+        # a log file is created for each plate or data set name that holds all outputs and errors
+        with open(
+            pathlib.Path(
+                f"logs/cellprofiler_output_{pathlib.Path(path_to_loaddata).name}.log"
+            ),
+            "w",
+        ) as cellprofiler_output_file:
+            # run CellProfiler for a IC or zproj pipeline
+            command = [
+                "cellprofiler",
+                "-c",
+                "-r",
+                "-p",
+                path_to_pipeline,
+                "-o",
+                path_to_output,
+                "--data-file",
+                path_to_loaddata,
+            ]
+            subprocess.run(command, capture_output=cellprofiler_output_file, check=True)
 
     if analysis_run:
         # runs through any files that are in the output path
         if any(
             files.name.startswith(sqlite_name)
             for files in pathlib.Path(path_to_output).iterdir()
         ):
-            raise NameError(f'The file {sqlite_name}.sqlite has already been renamed! This means it was probably already analyzed.')
+            raise NameError(
+                f"The file {sqlite_name}.sqlite has already been renamed! This means it was probably already analyzed."
+            )
 
         # run CellProfiler on a plate that has not been analyzed yet
-        command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} --data-file {path_to_loaddata}"
-        os.system(command)
+        print(f"Starting CellProfiler run on {pathlib.Path(path_to_loaddata).name}")
+        # a log file is created for each plate or data set name that holds all outputs and errors
+        with open(
+            pathlib.Path(
+                f"logs/cellprofiler_output_{pathlib.Path(path_to_loaddata).name}.log"
+            ),
+            "w",
+        ) as cellprofiler_output_file:
+            # run CellProfiler for an analysis run
+            command = [
```

Comment:
this is very similar to the command listed in line 79 - is it actually the same? Maybe not worth it now, but can you think of how to not copy and paste this? remember DRY! 😄

Reply:
Huh, it actually is the same! In the NF1 project, we don't use a LoadData CSV so there is a different command for IC pipeline and analysis pipeline. I can figure out a way to make this better, so I will add this as an issue!